src/legendhpges/base.py

```diff
@@ -55,19 +55,29 @@ def __init__(
         else:
             self.name = name
 
-        # return ordered r,z lists, default unit [mm]
-        r, z = self._decode_polycone_coord()
+        self.registry = registry
 
-        # build generic polycone, and logical volume, default [mm]
-        ic_solid = geant4.solid.GenericPolycone(
-            self.name, 0, 2 * math.pi, r, z, registry
-        )
-
-        super().__init__(ic_solid, material, self.name, registry)
+        # build logical volume, default [mm]
+        super().__init__(self._ic_solid(), material, self.name, registry)
 
     def __repr__(self) -> str:
         return f"{self.__class__.__name__}({self.metadata})"
 
+    @abstractmethod
+    def _ic_solid(self) -> geant4.solid.SolidBase:
```

Comment:
Why not providing a default implementation and only override it in `P00664B`?

Reply:
I think `ic` is because the first implementation was of an IC detector. Maybe now `_g4_solid` is a better name now? @ManuelHu?

Reply:
Yes, default implementation, then overrides would be nicer. What if we keep the name consistent with `_hpge_solid` for example? I think `_g4_solid` is too general. Also, from Tomasso's comment in [here](https://github.com/legend-exp/legend-detectors/pull/35#pullrequestreview-1474897086) (and cross-checking the datasheet), `V02160A` apparently also has a special geometry. I'll realize it later.

Reply:
Yes, we need a custom class for V02160A too!

I think `hpge` is redundant since the class is named `HPGe`.